internal/runner/operation_task.go

```diff
@@ -50,7 +52,60 @@ func (r *Runner) executeStartTaskOp(
 
 	fn := c.(component.TaskLauncher).StartTaskFunc()
 
-	_, err = pi.Invoke(ctx, log, fn, tli)
+	val, err := pi.Invoke(ctx, log, fn, tli)
+	if err != nil {
+		return nil, err
+	}
+
+	result, err := component.ProtoAny(val)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.Job_Result{
+		StartTask: &pb.Job_StartTaskResult{
+			State: result,
+		},
+	}, nil
+}
+
+func (r *Runner) executeStopTaskOp(
+	ctx context.Context,
+	log hclog.Logger,
+	ui terminal.UI,
+	job *pb.Job,
+) (*pb.Job_Result, error) {
+	op, ok := job.Operation.(*pb.Job_StopTask)
+	if !ok {
+		// this shouldn't happen since the call to this function is gated
+		// on the above type match.
+		panic("operation not expected type")
+	}
+
+	pi, c, err := plugin.OpenPlugin(ctx, log, &plugin.PluginRequest{
+		Config: plugin.Config{
+			Name: "dack",
+		},
+		Dir:  "/tmp",
+		Type: component.TaskLauncherType,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	defer pi.Close()
+
+	stop := c.(component.TaskLauncher).StopTaskFunc()
+
+	msg, err := ptypes.AnyMessageName(op.StopTask.State)
+	if err != nil {
+		// This should never happen.
+		panic(err)
+	}
+
+	arg := argmapper.NamedSubtype("state", op.StopTask.State, msg)
```

Comment:
Ah, there is a helper for this in `internal/core` I'm sure you found. I wonder if we should extract this. (internal/core/arg.go)

Reply:
Yup, that's where I pulled this code over from actually. Should we just export `argNamedAny` ?

Reply:
I think its weird for `core` to do this, makes a lot of sense imo to pull it into `internal/plugin` and export.

Reply:
I went ahead and exported it.

Reply:
I think it still feels out of scope for the "core" package to do this and more in line with the goals of the "plugin" package.

(Thinking through someone reading through a `godoc` I'd think... why?)

Reply:
I can move it over to plugin, sure.

Reply:
Ok, moved it over to plugin.